SharedPackages/AttributedMetric/Tests/AttributedMetricTests/RollingEightDaysIntTests.swift

```diff
@@ -0,0 +1,217 @@
+    }
+
+    func testPast7DaysAverageWithValues() {
+        // Add values to fill some slots (including today)
+        for i in 1...8 {
+            rollingInt.append(i)
+        }
+
+        // past7DaysAverage should exclude the last value (today)
+        // Values: [1, 2, 3, 4, 5, 6, 7], average = (1+2+3+4+5+6+7)/7 = 4
+        let expectedAverage = Int((Float(1+2+3+4+5+6+7) / Float(rollingInt.count-1)).rounded(.toNearestOrAwayFromZero))
+        XCTAssertEqual(rollingInt.past7DaysAverage, expectedAverage)
+    }
+
+    // Discussion still ongoing, the average will need to be adjusted
+
+//    func testPast7DaysAverageWithUnknownValues() {
+//        // Add some values and leave some unknown
+//        rollingInt.append(3)
+//        rollingInt.append(7)
+//        rollingInt.append(11)
+//        rollingInt.append(0)
+//
+//        // past7DaysAverage should only count known values (excluding today)
+//        // Values excluding last: [10, 20, 30], average = (3+7+11)/2 = 10.5
+//        XCTAssertEqual(rollingInt.past7DaysAverage, 11)
+//    }
+//
+//    func testPast7DaysAverageRoundingBehavior() {
+//        // Test specific rounding cases
+//        rollingInt.append(1)  // Will be excluded (today)
+//        rollingInt[0] = 6     // 6
+//        rollingInt[1] = 7     // 7
+//
+//        // Average = (6+7)/2 = 6.5, rounded = 7
+//        XCTAssertEqual(rollingInt.past7DaysAverage, 7)
+//
+//        // Test rounding down case
+//        rollingInt[2] = 5     // 5
+//
+//        // Average = (6+7+5)/3 = 6, no rounding needed
+//        XCTAssertEqual(rollingInt.past7DaysAverage, 6)
+//    }
+
+    func testCountPast7DaysEmptyArray() {
+        XCTAssertEqual(rollingInt.countPast7Days, 0)
+    }
+
+    func testCountPast7DaysWithValues() {
+        // Fill array with values
+        for i in 1...8 {
+            rollingInt.append(i)
+        }
+
+        // Should count all non-unknown values excluding today (last value)
+        // Total values = 8, excluding today = 7
+        XCTAssertEqual(rollingInt.countPast7Days, 7)
+    }
+
+    func testCountPast7DaysWithMixedValues() {
+        // Add some values, leave some unknown
+        rollingInt.append(1)
+        rollingInt.append(2)
+        rollingInt[5] = 99  // Manually set a value
+
+        // Should count only non-unknown values excluding today
+        // Values excluding last: [unknown, unknown, unknown, unknown, unknown, 99, 1] = 2 non-unknown
+        XCTAssertEqual(rollingInt.countPast7Days, 2)
+    }
+
+    func testMultipleDaysSequenceWithIncrements() {
+        let currentDate = Date()
+
+        // Simulate multiple days with different increment counts
+        let dailyIncrements = [3, 1, 5, 2, 4, 1, 3, 2, 1]
+
+        for (dayIndex, increments) in dailyIncrements.enumerated() {
+            // Set lastDay to simulate different days
+            if dayIndex > 0 {
+//                currentDate = Calendar.current.date(byAdding: .day, value: 1, to: currentDate)!
```

Comment:
Any reason to keep the commented out line?

Reply:
removed